opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -584,6 +588,42 @@ namespace Opm
         }
     }
 
+    template<typename TypeTag>
+    bool
+    WellInterface<TypeTag>::
+    gliftBeginTimeStepWellTestIterateWellEquations(
+        const Simulator& ebos_simulator,
+        const double dt,
+        WellState& well_state,
+        const GroupState &group_state,
+        DeferredLogger& deferred_logger)
+    {
+        const auto& well_name = this->name();
+        const auto& summary_state = ebos_simulator.vanguard().summaryState();
+        assert(this->wellHasTHPConstraints(summary_state));
+        const auto& schedule = ebos_simulator.vanguard().schedule();
+        auto report_step_idx = ebos_simulator.episodeIndex();
+        const auto& glo = schedule.glo(report_step_idx);
+        if(glo.has_well(well_name)) {
+            auto increment = glo.gaslift_increment();
+            auto alq = well_state.getALQ(well_name);
+            bool converged;
+            while (alq > 0) {
+                well_state.setALQ(well_name, alq);
+                if ((converged =
+                      iterateWellEquations(ebos_simulator, dt, well_state, group_state, deferred_logger)))
+                {
+                    return converged;
+                }
+                alq -= increment;
+            }
```

Comment:
Would it be better to start iterating from alq = 0.0 and increase it until either convergence or maximum alq rather than the opposite.

Reply:
Thanks for the comment. Than the code is fine.